xbmc/guilib/StereoscopicsManager.cpp

```diff
@@ -99,29 +99,38 @@ static const struct StereoModeMap StringToGuiModeMap[] =
 };
 
 
-CStereoscopicsManager::CStereoscopicsManager(void)
+CStereoscopicsManager::CStereoscopicsManager(CSettings &settings,
+                                             CDataCacheCore &dataCacheCore,
+                                             CGUIWindowManager &windowManager) :
+  m_settings(settings),
+  m_dataCacheCore(dataCacheCore)
 {
   m_stereoModeSetByUser = RENDER_STEREO_MODE_UNDEFINED;
   m_lastStereoModeSetByUser = RENDER_STEREO_MODE_UNDEFINED;
-}
 
-CStereoscopicsManager::~CStereoscopicsManager(void) = default;
+  std::set<std::string> settingSet{
+    CSettings::SETTING_VIDEOSCREEN_STEREOSCOPICMODE
+  };
+
+  m_settings.GetSettingsManager()->RegisterCallback(this, settingSet);
```

Comment:
better do this in Initialize(). avoiding behaviour in ctor also avoids potential problems

Reply:
Good call. I'm going to mark this as a TODO because our CGUIWindowLoginScreen::LoadProfile() nightmare smashes through virtually every boundary. In this case, it double-initializes a bunch of stuff on profile change, including CStereoscopicsManager: https://github.com/xbmc/xbmc/blob/master/xbmc/windows/GUIWindowLoginScreen.cpp#L356. Better to clearly mark theoretical problems for now instead of introducing guaranteed ones :)